### PR TITLE
Handle exception thrown in log while trying to call sun.misc.VM.maxDirectMemory() which is not available in Java 11

### PR DESCRIPTION
**Pull request by dclim**
Part of #5589, was missed during #7606

### CODE REVIEW

File: services/src/main/java/org/apache/druid/cli/GuiceRunnable.java [hunk 1]
```diff
@@ -78,12 +78,20 @@ public Lifecycle initLifecycle(Injector injector)
...
+      }
+
       log.info(
-          "Starting up with processors[%,d], memory[%,d], maxMemory[%,d], directMemory[%,d].",
+          "Starting up with processors[%,d], memory[%,d], maxMemory[%,d]%s.",
           JvmUtils.getRuntimeInfo().getAvailableProcessors(),
           JvmUtils.getRuntimeInfo().getTotalHeapSizeBytes(),
           JvmUtils.getRuntimeInfo().getMaxHeapSizeBytes(),
-          JvmUtils.getRuntimeInfo().getDirectMemorySizeBytes()
+          directSizeBytes != null ? String.format(", directMemory[%,d]", directSizeBytes) : ""
```
Comment (clintropolis) [inline services/src/main/java/org/apache/druid/cli/GuiceRunnable.java]: Ah, this should use `StringUtils.format` instead of `String.format`.

Comment (dclim) [reply services/src/main/java/org/apache/druid/cli/GuiceRunnable.java]: Oops, thanks Clint

**Comment by gianm** [comment]
LGTM, I'll merge even though Coveralls is complaining, because the tool seems to be misbehaving (there is no way this change decreased coverage by 9.1%).

**Comment by xvrl** [comment]
Thanks for catching this. Is it possible none of our test run this code? I thought I had run most unit tests.
Might be worth adding a test if this code is not covered already.

**Pull request merged**